mypy/checker.py

```diff
@@ -127,8 +128,12 @@ class TypeChecker(NodeVisitor[None]):
     # directly or indirectly.
     module_refs = None  # type: Set[str]
 
+    # Plugin that provides special type checking rules for specific library
+    # functions such as open(), etc.
+    plugin = None  # type: Plugin
+
     def __init__(self, errors: Errors, modules: Dict[str, MypyFile], options: Options,
-                 tree: MypyFile, path: str) -> None:
+                 tree: MypyFile, path: str, plugin: Optional[Plugin] = None) -> None:
```

Comment:
Hmm probably not. It's trivial to provide an instance of the empty `Plugin` as the argument in case no plugin functionality is needed. I'll make this non-optional if it doesn't break anything.